### PR TITLE
[codex] Harden lookup and normalization padding guards

### PR DESCRIPTION
**Pull request by omarespejel**
## Summary
- replace lookup and normalization table padding `expect(...)` paths with explicit `VmError::InvalidConfig` guards
- propagate the new fallible demo-bundle builders through the Phase3/5 demo prove/verify paths
- add direct fail-closed tests for empty canonical lookup/normalization row inputs

## Validation
- `cargo +nightly-2025-07-14 test --features stwo-backend --lib 'stwo_backend::lookup_prover::tests'`
- `cargo +nightly-2025-07-14 test --features stwo-backend --lib 'stwo_backend::normalization_prover::tests'`
- `cargo +nightly-2025-07-14 fmt --check`
- `cargo +nightly-2025-07-14 check --features stwo-backend --lib --bin tvm`
- `bash scripts/run_tablero_hardening_preflight.sh --mode core`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling for lookup and normalization table operations; invalid configurations now return structured errors instead of panicking.

* **Tests**
  * Added validation tests to ensure empty canonical rows are properly rejected with appropriate error messages.

### CODE REVIEW

File: src/stwo_backend/lookup_prover.rs [hunk 1]
```diff
@@ -113,7 +113,7 @@ pub struct Phase10SharedLookupProofEnvelope {
 }
 
 pub fn prove_phase3_binary_step_lookup_demo_envelope() -> Result<Phase3LookupProofEnvelope> {
-    let bundle = build_lookup_demo_bundle();
+    let bundle = build_lookup_demo_bundle()?;
```
[the remark this answers is not in the document]
Comment (omarespejel) [reply src/stwo_backend/lookup_prover.rs:116]: Ran the requested checks and classified the failures before merge:

- `just gate-fast`: completed successfully before the earlier chained run advanced to `just lib`
- `just lib`: fails on this branch, but the same representative release-mode decoding failures reproduce unchanged on a clean detached worktree at the PR base commit `c3de15333e389db76ef05f4e290cf371caf2520d`
  - `stwo_backend::decoding::tests::load_phase14_decoding_chain_rejects_empty_public_state_commitment`
  - `stwo_backend::decoding::tests::phase14_prepare_decoding_chain_accepts_linked_steps`
  - `stwo_backend::decoding::tests::phase12_oracle_matches_production_on_demo_chain`
  - all fail with the same pre-existing `UnsupportedProof("proof claim uses statement-v1 semantic scope but is missing the transformer configuration required by that scope; reject as claim-drift")`
- `cargo +nightly-2025-07-14 clippy --lib --no-deps -- -D warnings`: currently fails only in untouched `src/tui.rs` on pre-existing `clippy::uninlined_format_args` findings at lines 637, 730, 1005, and 1139
- `just gate` is therefore blocked by repo-baseline failures outside this PR slice, not by the lookup/normalization guard changes

PR-local validation that did pass is unchanged:
- `cargo +nightly-2025-07-14 test --features stwo-backend --lib 'stwo_backend::lookup_prover::tests'`
- `cargo +nightly-2025-07-14 test --features stwo-backend --lib 'stwo_backend::normalization_prover::tests'`
- `cargo +nightly-2025-07-14 fmt --check`
- `cargo +nightly-2025-07-14 check --features stwo-backend --lib --bin tvm`
- `bash scripts/run_tablero_hardening_preflight.sh --mode core`

Given the baseline classification, resolving this as non-blocking for this narrowly scoped PR.
